### PR TITLE
Branch dg appendix

### PR DESCRIPTION
**Pull request by ldinghan**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1226,6 +1226,84 @@ testers are expected to do more *exploratory* testing.
...
 
+
+### Creating a session
+
+1. Creating a session with a new session number
+    
+    1. Prerequisites: Student names must exist in address book and no session with the session number given exists.
+    1. Test case: `createsession s/2 n/Alex Yeoh`<br>
+       Expected: A session with session number 2, containing the student Alex Yeoh, is created and stored in the list of sessions.
+    1. Test case: `createsession s/2 n/Alex Yeoh n/Bernice Yu`<br>
```
Comment (raydenlim) [inline docs/DeveloperGuide.md:1237]: why is everything indented as 1. Change it to down down per test case

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1226,6 +1226,84 @@ testers are expected to do more *exploratory* testing.
...
+        Expected: A session with session number 2, containing the students Alex Yeoh and Bernice Yu, is created and stored in the list of sessions.
+    1. Test case: `createsession s/2 n/Alex Yeoh` `createsession s/2 n/Alex Yeoh`<br>
+       Expected: No new session is created on the second command. Error details shown in the status message.
+    1. Other incorrect create session commands to try: `createsession`, `createsession n/Alex Yeoh`<br>
+        Expected: Similar to previous
+
+### Updating a session remark
+
+1. Updating the remark field of a session
```
Comment (raydenlim) [inline docs/DeveloperGuide.md:1246]: same as above

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1226,6 +1226,84 @@ testers are expected to do more *exploratory* testing.
...
+        Expected: No update to any session's remarks. Error details shown in the status message.
+   1. Other incorrect update session remark commands to try: `updatesessionremark`, `updatesessionremark s/1`
+        Expected: Similar to previous.
+
+
+### Deleting a session
+
+1. Deleting a session
+    1. Prerequisites: Session must exist in the session list.
+    1. Test case: `deletesession s/2`<br>
```
Comment (raydenlim) [inline docs/DeveloperGuide.md:1260]: same as above

**Review by raydenlim — COMMENTED** [review]
My bad fam, i thought its a mistake. Turns out, it looks like a convention. Other than that, lgtm

**Review by Originalidk — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1910,6 +1912,144 @@ testers are expected to do more *exploratory* testing.
...
+
+
+#### Updating a session remark
+
+1. Updating a session remark with _valid parameters_
+  1. Prerequisites: Session must exist in the session list.
+  1. Test case: `updatesessionremark s/2 r/Update the remark to this text`<br>
+     Expected: The session with session number 2 has its remark field updated to "Update the remark to this text".
+
+1. Updating a session remark with _missing parameters_
```
Comment (raydenlim) [inline docs/DeveloperGuide.md]: might want to standardise, parameters/arguments.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1910,6 +1912,144 @@ testers are expected to do more *exploratory* testing.
...
+
+1. Taking attendance of a _valid student_
+  1. Prerequisites: Session must exist in the session list. Student name must exist in address book.
+  1. Test case: `takeattendance s/2 n/David Li ap/present`<br>
+     Expected: David Li is added to the session with session number 2, marking him as present.
+
+1. Taking attendance of an _invalid student_
+  1. Test case: `takeattendance s/2 n/fakenamethatdoesnotexist ap/absent`<br>
+     Expected: F.A.K.E.J.A.R.V.I.S. displays an error. No attendance is taken. <br>
+     Reason: Student name must exist in address book.
```
Comment (raydenlim) [inline docs/DeveloperGuide.md]: Student name must be in the StudentList?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1910,6 +1912,144 @@ testers are expected to do more *exploratory* testing.
...
+     Reason: Attendance presence argument is missing.
+  1. Test case: `takeattendance s/2 ap/present`<br>
+     Expected: F.A.K.E.J.A.R.V.I.S. displays an error. No attendance is taken. <br>
+     Reason: Student name argument is missing.
+  1. Test case: `takeattendance`<br>
+     Expected: F.A.K.E.J.A.R.V.I.S. displays an error. No attendance is taken. <br>
+     Reason: Session number, student name and attendance presence arguments are missing.
+
+1. Taking attendance of _multiple valid students_
+  1. Prerequisites: Session must exist in the session list. Student names must exist in address book.
```
Comment (raydenlim) [inline docs/DeveloperGuide.md]: Student List

**Review by raydenlim — APPROVED** [review]
LGTM apart from some nits

File: docs/UserGuide.md [hunk 7]
```diff
@@ -324,35 +324,35 @@ The following describes what each tab displays on the **Current Panel**:
...
-|     GRADED_TEST      |   `gt/`    |                                                   `add` `edit`                                                    | The Scores of respective Graded Tests. <ul><li>Graded Test should only contain alphanumeric characters and spaces. </li><li>Graded Test field should follow this format: <code>gt/RA1:&lt;SCORE> &#124; RA2:&lt;SCORE> &#124; MidTerms:&lt;SCORE> &#124; Finals:&lt;SCORE> &#124; PE:&lt;SCORE></code>, where `<SCORE>` **must be a positive float**. </li></ul> |
-| READING_ASSESSMENT_1 |   `ra1/`   |                                                 `editgradedtest`                                                  | The Score of Reading Assessment 1. <ul><li>Score should be a positive number.</li></ul>                                                                                                                                                                                                                                                                          |
```
Comment (raydenlim) [inline docs/UserGuide.md:335]: good catch!

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -344,7 +344,7 @@ GradedTest testFromObjects = new GradedTest(
 - Cons:
   * Testing Overhead: The implementation of both alternatives increases testing complexity, requiring thorough testing to ensure compatibility and proper functionality.
 
-- For the UML diagram of `EditGradedTest` refer to [Edit Graded Test](#).
+- For the UML diagram of `EditGradedTest` refer to [Edit Graded Test](#edit-graded-test-feature).
```
Comment (raydenlim) [inline docs/DeveloperGuide.md:347]: good catch!

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -736,18 +736,10 @@ The `EditGradedTestCommand` will call its own `updateFilteredPersonList` method
 Step 5:
 The `EditGradedTestCommand` then continues its execution as defined by [this](#parser-commands) sequence diagram.
 
-
-<div class="alert alert-info"> 
-<md>
-
-:information_source: **Note**:
-
 * The process of `createEditedGradedTestPerson` is summarised in the activity diagram below:
```
Comment (raydenlim) [inline docs/DeveloperGuide.md:739]: Oops, my bad

**Review by raydenlim — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (ldinghan) [reply docs/DeveloperGuide.md]: alright, standardise with the rest to make it parameters

**Pull request merged**